lib/utils.js

```diff
@@ -505,6 +505,26 @@ function isPromiseLike(maybePromise) {
   return maybePromise && typeof maybePromise.then === 'function';
 }
 
+/**
+ * Applies collation to a given command.
+ *
+ * @param {object} [command] the command on which to apply collation
+ * @param {(Cursor|Collection)} [self]
+ * @param {object} [options] options containing collation settings
+ */
+function decorateWithCollation(command, self, options) {
+  if (self.s && self.s.topology) {
```

Comment:
I feel like this is potentially hiding issues? If we happen to call this function with something that doesn't have a topology, we silently fail to do anything. In my opinion we should throw an error here: fail fast and hard